### DISCUSSION TITLE
Edge TTS Plugin Stopped Working

### DISCUSSION THREAD

**Discussion by mark0mark0mark** [travisvn/obsidian-edge-tts#13] [General]
I had the Edge TTS working for a few days just fined.  But a couple of days ago, it stopped working.
I have the latest version of Obsidian.  I uninstalled Edge TTS and reinstalled it.  I updated it.  I restarted Obsidian.  But it's still not working.
I see this in the Developer's Console:

plugin:edge-tts:34 Uncaught (in promise) DOMException: Failed to execute 'stop' on 'AudioScheduledSourceNode': cannot call stop without calling start first.
    at P.stopPlayback (plugin:edge-tts:34:4289)
    at P.readNoteAloud (plugin:edge-tts:34:2477)
    at Object.editorCallback (plugin:edge-tts:32:2229)
    at e.mobileOnly.Pl.isMobile.e.checkCallback (app://obsidian.md/app.js:1:1936161)
    at ZU (app://obsidian.md/app.js:1:1935421)
    at t.onChooseItem (app://obsidian.md/app.js:1:2544224)
    at t.onChooseSuggestion (app://obsidian.md/app.js:1:1761296)
    at t.selectSuggestion (app://obsidian.md/app.js:1:1760766)
    at e.useSelectedItem (app://obsidian.md/app.js:1:1434310)
    at Object.func (app://obsidian.md/app.js:1:1431709)

Any suggestions?

Thanks,

Mark

**Comment by GyckaZ** [comment]
I am very sorry, but generate .mp3 function seems not to work. I am very excited and look forward to try this function once it is fixed. Currently, the new file is generated when I click on generate .mp3, but the file what is generated does not contain .mp3 file extension. If I add the extension manually and try to open it, it says the file has unsupported format or it is corrupted. Otherwise, the plugin works great.

**Reply by travisvn** [reply]
> I am very sorry, but generate .mp3 function seems not to work. I am very excited and look forward to try this function once it is fixed. Currently, the new file is generated when I click on generate .mp3, but the file what is generated does not contain .mp3 file extension. If I add the extension manually and try to open it, it says the file has unsupported format or it is corrupted. Otherwise, the plugin works great.

Hi @GyckaZ - I need more information to be able to help you, as it works on my setup. 

What operating system are you using, and can you provide any of the console logs that can be found in the View -> Toggle Developer Tools menu?

**Reply by GyckaZ** [reply]
Hi there :)

I am using MS Windows 11 Pro
![image](https://github.com/user-attachments/assets/9fabb88f-4143-4098-8c0a-fc411d25f94f)

Adding a screenshot of the console, I cant seem to find anything Edge TTS related here, anything specific I should look for?

![image](https://github.com/user-attachments/assets/2bc8e7cb-9d3e-4428-be57-e8e4ea59cf8c)

Also, the generated file does nto have .mp3 extension and its size is 0 bytes, maybe this can indicate where the problem is?
![image](https://github.com/user-attachments/assets/8d35fb53-5169-4fd2-80d9-42817c0edebc)

**Comment by travisvn** [comment]
> I had the Edge TTS working for a few days just fined. But a couple of days ago, it stopped working. I have the latest version of Obsidian. I uninstalled Edge TTS and reinstalled it. I updated it. I restarted Obsidian. But it's still not working. I see this in the Developer's Console:
> 
> plugin:edge-tts:34 Uncaught (in promise) DOMException: Failed to execute 'stop' on 'AudioScheduledSourceNode': cannot call stop without calling start first. at P.stopPlayback (plugin:edge-tts:34:4289) at P.readNoteAloud (plugin:edge-tts:34:2477) at Object.editorCallback (plugin:edge-tts:32:2229) at e.mobileOnly.Pl.isMobile.e.checkCallback (app://obsidian.md/app.js:1:1936161) at ZU (app://obsidian.md/app.js:1:1935421) at t.onChooseItem (app://obsidian.md/app.js:1:2544224) at t.onChooseSuggestion (app://obsidian.md/app.js:1:1761296) at t.selectSuggestion (app://obsidian.md/app.js:1:1760766) at e.useSelectedItem (app://obsidian.md/app.js:1:1434310) at Object.func (app://obsidian.md/app.js:1:1431709)
> 
> Any suggestions?
> 
> Thanks,
> 
> Mark

Hey @mark0mark0mark - What operating system are you using? And do any of the functions for reading aloud work? 

(Generating an MP3, right clicking on a file to select `Read note aloud` or `Generate mp3`, selecting text in your note and then clicking the button in the ribbon or status bar, etc.)

**Reply by mark0mark0mark** [reply]
I am using Windows 11.  I don't know what happened, but it's working now.  I had installed the Speechify extension on my Edge browser a couple of days ago.  So, I uninstalled it.  That didn't fix my Obsidian problem.  Installed the Speechify extension in Edge and got that working.  Then your Edge-TTS started working too. I don't know if Speechify had anything to do with it. Sorry for the false alarm.

**Reply by travisvn** [reply]
> I am using Windows 11. I don't know what happened, but it's working now. I had installed the Speechify extension on my Edge browser a couple of days ago. So, I uninstalled it. That didn't fix my Obsidian problem. Installed the Speechify extension in Edge and got that working. Then your Edge-TTS started working too. I don't know if Speechify had anything to do with it. Sorry for the false alarm.

All good — glad it's working! 

Have you liked Speechify? I was considering a subscription with them a couple years ago (before there were TTS offerings like Edge provides)

**Comment by mark0mark0mark** [comment]
I got a subscription to Speechify because at first glance it seemed to work
pretty well.  But the main reason I got it is to read my Kindle books to
me, and I discovered that it has some flaws in that it sometimes gets part
way through a chapter and skips back and continues reading what it has
already read.  I have a couple of times been stuck in a chapter where it
won't finish the chapter--it just keeps skipping backwards.

I'm documenting this for their Support time right now.

I wouldn't recommend it, if you need to read Kindle books.  I haven't had
that problem elsewhere, but I supposed the web pages I've read haven't been
that long.
